hookbook.sh

```diff
@@ -130,11 +128,14 @@ __hookbook_call_each() {
     __hookbook_array_contains "${fn}" "${__hookbook_functions[@]}" \
       || __hookbook_functions+=("${fn}")
   }
+}
+
+[[ "${__hookbook_shellname}" != "zsh" ]] && [[ "${__hookbook_shellname}" != "bash" ]] && {
+  >&2 \echo "hookbook is not compatible with your shell (${__hookbook_shell})"
   \unset __hookbook_shell __hookbook_shellname
-  \return 0
+  \return 1
 }
 
->&2 \echo "hookbook is not compatible with your shell (${__hookbook_shell})"
-\return 1
+\unset __hookbook_shell __hookbook_shellname
```

Comment:
It looks like you want to call this regardless of the conditional on 133; could you move it above and remove it from the conditional body?

Reply:
No - `__hookbook_shell` is used on 134, but should be removed before 136 (and generally).